database/etcd.go

```diff
@@ -0,0 +1,142 @@
+package database
+
+import (
+	"fmt"
+	"path"
+	"strconv"
+	"strings"
+
+	"github.com/gobackup/gobackup/helper"
+	"github.com/gobackup/gobackup/logger"
+)
+
+// etcd database
+//
+// ref:
+// https://etcd.io/docs/v3.4/dev-guide/interacting_v3/
+//
+// # keys
+//
+//   - type: etcd
+//   - endpoints: [localhost]
+//   - port: 2379
+//   - user:
+//   - password:
+//   - cacert:
+//   - cert:
+//   - key:
+//   - insecure-skip-tls-verify: false
+//   - args:
+type Etcd struct {
+	Base
+	endpoints             []string
+	user                  string
+	password              string
+	caCert                string
+	cert                  string
```

Comment:
Is the `cert`, `caCert` and `key` is a requirement? 
If not, I'd like to leave this to use `args`, too many options will make GoBackup hard for maintaining and testing.

Reply:
No these args are not required, but in most cases of deploying etcd in kubernetes these args are required.
Also neither username nor password are required arguments for etcd.